data/publiccloud/azure_vn.sh

```diff
@@ -0,0 +1,420 @@
+#!/bin/bash -u
+set -o pipefail
+. $(dirname "${BASH_SOURCE[0]}")/azure_lib_fn.sh
+#set -x
+# Create a VM with multiple subnets and public ip
+# maximize the usage of azure cli network
+# list, show , delete the created resource
+
+# Virtual Network AZURE CLI testing script
+
+#Required AZ setting variable block
+grpname="${1:-oqaclitest}"
+location="${2:-westus}"
+vmname="${3:-oqaclivm}"
+ssh_key="${4:-oqaclitest-sshkey}"
+vmximagename="${5:-UbuntuLTS}"
+
+# Variable block
+account="Azure RD"
+admin="azureuser"
+stor="stor"
+cont="cont"
+storageacc=$grpname$stor
+STORAGE_CONT=$grpname$cont
+number_of_nics=2
+addressprefix="30.0.0.0/16"
+PrivateIpAddress="30.0.0.10"
+PrivateIpAddress2="30.0.1.10"
+subnet_prefixes=( 30.0.0.0/24 30.0.1.0/24 )
+#vmximagename="oqacli-image-20230127175309"
+vnet=$grpname-vnet1
+base_subnet=$vnet-vsubnet
+base_nic=$vmname-wan-nic
+mdisk="_Managed_Disk"
+echo "**Start of AZURE CLI Virtual Network test**"
+echo "*******************************************"
+
+# Resource creation
+echo "Set Azure Account { az account set }"
+cmd_status "az_account_set" az account set -n "$account"
+
+echo "Azure account set to $account"
+
+echo "Creating resource group { az group create }"
+cmd_status "az_group_create" az group create -n "$grpname" -l "$location" -o table
+echo "Created group $grpname"
+
+echo "Creating storage account { az storage account create }"
+# Create storage account
+cmd_status "az_storage_account_create" az storage account create --sku Standard_LRS --location "$location" --kind Storage --resource-group "$grpname" --name "$storageacc" --output table
+echo "Created storage account $storageacc"
+
+# Get connection string for the storage account 
+echo "Get Connection string for storage account { az storage account keys list }"
+cmd_status "az_storage_account_keys_list" az storage account keys list --account-name "$storageacc" --resource-group "$grpname"
+KEY1=$(az storage account keys list --account-name $storageacc --resource-group $grpname  | grep -m 1 \"value\": | awk '{print $2}')
+
+# Create storage container
+echo "Create storage container { az storage container create }"
+cmd_status "az_storage_container_create" az storage container create --account-name "$storageacc" --name "$STORAGE_CONT" --account-key "$KEY1" -o table
+echo "Created container $STORAGE_CONT"
+
+count=$(( $number_of_nics - 1 ))
+while [ $count -ge 0 ]
+do
+    subnet_names[$count]=$base_subnet"$count"
+    nic_names[$count]=$base_nic"$count"
+    count=$(( $count - 1 ))
+done
+
+skip=1
```

Comment:
can you explain why do you need this skip variable ?

Reply:
done. I have refactored azure_vn.sh  

Intention initially is to check vnet already exists, if not create vnet and subnet